network/p2p/router.go

```diff
@@ -195,7 +195,6 @@ func (r *Router) RegisterAppProtocol(handlerID uint64, handler Handler, nodeSamp
 		handlerPrefix: binary.AppendUvarint(nil, handlerID),
```

Comment:
Why is the client constructed by `newAppProtocol`? Is it just because the router knows about how `handlerPrefix` is defined better than the network does? I feel like constructing the `Client` in `Network.NewAppProtocol` would make a bit more sense than here (and the network already has everything it needs to create this client)

Reply:
Purely for legacy reasons. I think this does look much cleaner to just move this into the `Network`. I will move the client stuff.